src/coreclr/jit/morph.cpp

```diff
@@ -9652,11 +9652,7 @@ GenTree* Compiler::fgMorphBlockOperand(GenTree* tree, var_types asgType, unsigne
 //    lclNum2 - the local variable on the other side of ASG, can be BAD_VAR_NUM.
 //
 // Return Value:
-//    True if the second local is valid and has the same struct handle as the first,
-//    false otherwise.
-//
-// Notes:
-//   TODO-PhysicalVN: with the physical VN scheme, this method is no longer needed.
+//    True if the second local is valid and has a layout compatible with the first local.
```

Comment:
The `TODO` should stay or the whole method deleted (what #68986 does).

There is no reason we cannot use `LCL_FLD` in a construct such as the following:
```c
lcl.StructField = promotedLocal;
```

Reply:
Sorry, I have not noticed that #68986 removes this, I will just leave it for that PR then.

> There is no reason we cannot use LCL_FLD<lcl> in a construct such as the following:

I see, makes sense. My main worry here was about dropping GC ness during the copy, but I guess the alternatives do not do anything different in that regard.